### PR TITLE
feat(install-local.sh): add `DCPUFLAGS` + `cpuflags` arrays

### PR DESCRIPTION
**Pull request by oklopfer**
## Purpose
adds internal cpuflag check for making a pacscript rely on it or choose a package based on certain flags

## Approach

add 2 new arrays

## Progress

- [x] Do it
- [x] Test it

## Checklist

- [x] I confirm that I have read the [contributing guidelines](https://github.com/pacstall/pacstall/blob/develop/CONTRIBUTING.md), and this pull request is abiding by all the clauses stated in the guideline.

### CODE REVIEW

File: misc/scripts/install-local.sh [hunk 1]
```diff
@@ -271,6 +271,25 @@ function is_compatible_arch() {
...
+    local input=("${@}") ret=0
+    for pcpuflag in "${input[@]}"; do
+        local flag_found=0
+        for dcpuflag in "${DCPUFLAGS[@]}"; do
+            if [[ "$pcpuflag" == "$dcpuflag" ]]; then
+                flag_found=1
+                break
+            fi
+        done
+        if ((flag_found==0)); then
```
Comment (Elsie19) [inline misc/scripts/install-local.sh:284]:
```suggestion
        if ((flag_found)); then
```

**Review by Elsie19 — CHANGES_REQUESTED** [review]

File: misc/scripts/install-local.sh (hunk 1 above)
Comment (oklopfer) [reply misc/scripts/install-local.sh:284]: Nope. Not correct.

Comment (Elsie19) [reply misc/scripts/install-local.sh:284]: Why not.

Comment (oklopfer) [reply misc/scripts/install-local.sh:284]: This would be checking for the existence of the variable, not the value of it, 4head.

Comment (Elsie19) [reply misc/scripts/install-local.sh:284]: no it wouldn't... `0` evaluates to success and `1` to fail, then put that in an if statement and it's checking exit codes.

Comment (oklopfer) [reply misc/scripts/install-local.sh:284]: Except it doesn’t. Try it yourself.

Comment (Elsie19) [reply misc/scripts/install-local.sh:284]: image

Comment (Elsie19) [reply misc/scripts/install-local.sh:284]: Fine I concede bash is being stupid.

**Pull request closed without merging**